internal/jimm/role/role.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2024 Canonical.
+package role
+
+import (
+	"context"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	jimmnames "github.com/canonical/jimm/v3/pkg/names"
+)
+
+// RoleStore defines a storage interface to persist roles.
+type RoleStore interface {
```

Comment:
Yup, which is really how we should be using interfaces. To consume structs.

Reply:
I'm not sold on using an interface just to limit the scope of a struct. Say we passed in a DB struct instead of a RoleStore. Then the RoleManager _could_ do more things. 
Is it worth it to have an interface to clearly describe what DB operations the RoleManager has access to?
I lean towards yes, but it also feels okay to provide the role manager with raw DB access.
This interface basically feels like a form of documentation (and potentially lets you test with a fake DB if desired).